src/usb_moded-network.c

```diff
@@ -299,18 +300,16 @@ static int resolv_conf_dns(struct ipforward_data *ipforward)
 
 static int checklink(void)
 {
-   char *dest = malloc( 32 * sizeof(char)) ;
-   int ret = 0;
-
-   readlink("/etc/udhcpd.conf", dest, 32);
-   if(dest != NULL)
-   {
-	strcpy(&dest[31], "\0");
-	ret = (strcmp(dest, UDHCP_CONFIG_PATH));
-   }
-   free(dest);
-
-   return(ret);
+  int ret = -1;
+  const size_t bufsiz = 32;
+  char *dest = malloc(bufsiz);
```

Comment:
If you decide to change that, note that sizeof(UDHCP_CONFIG_LINK) == sizeof(char*) -> fail.

Switching from macro to string variable would make the math work, but feels like overkill and requires more brain capacity to grok than some reasonably large buffer from stack.